packages/connection-form/src/components/advanced-options-tabs/csfle-tab/kms-provider-content.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useCallback, useMemo } from 'react';
+import {
+  Button,
+  Card,
+  css,
+  Icon,
+  IconButton,
+  spacing,
+  useHoverState,
+} from '@mongodb-js/compass-components';
+
+import type { UpdateConnectionFormField } from '../../../hooks/use-connect-form';
+import type {
+  KMSField,
+  KMSProviderType,
+  KMSProviderName,
+} from '../../../utils/csfle-kms-fields';
+import type { ConnectionFormError } from '../../../utils/validation';
+import type { ConnectionOptions } from 'mongodb-data-service';
+import KMSProviderFieldsForm from './kms-provider-fields';
+
+const cardStyles = css({
+  marginTop: spacing[200],
+  marginBottom: spacing[200],
+});
+
+const flexContainerStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+});
+
+const pushRightStyles = css({
+  marginLeft: 'auto',
+});
+
+function getNextKmsProviderName<T extends KMSProviderType>(
+  kmsProviderType: T,
+  currentProviders: string[]
+): KMSProviderName<T> {
+  const currentNums = currentProviders
+    .map((name) => parseInt(name.split(':')[1], 10))
+    .filter((num) => !isNaN(num));
+  const name = currentNums.length === 0 ? 1 : Math.max(...currentNums) + 1;
+  return `${kmsProviderType}:${name}`;
```

Comment:
I don't know if we can select the name for the user here. The name _is_ persisted in the key vault collection, and while I'm not entirely sure, I would guess that it has to match between different usages of the same key (i.e. the name used by the application `kmsProviders` config should match the one used here).

I wanted to verify this myself but ran into some issues with my local FLE setup that I'll probably need to look into first

Reply:
I updated the PR to add support for renaming kms. Currently it allows users to edit the name while warning them about:
1. Empty name
2. Duplicate name

In these cases, if a user proceeds to connect, currently it considers the last valid name (as we do not rename). 

It hides the full name of the provider (the prefix part) - e.g if the user types local123, the actual name being `local:local123`, we hide the `local:`. I am waiting design input on how to communicate this to the user.